src/Lean/Server.lean

```diff
@@ -133,10 +130,13 @@ writeLspNotification "textDocument/publishDiagnostics"
 
 def handleDidOpen (p : DidOpenTextDocumentParams) : ServerM Unit := do
 let doc := p.textDocument;
--- The text being split here is going to be immediately
--- intercalated with '\n' but this is useful to get rid of
--- CRLFs.
-let text := doc.text.splitOnEOLs.toArray;
+-- LSP is EOL agnostic, so we first split on "\r\n" to avoid possibly redundant line breaks.
```

Comment:
Shouldn't we rather fix this in `FileMap`? Actually, since no platform uses just `\r` anymore, `FileMap`'s approach of splitting after `\n` seems sufficient to me?

Reply:
You're possibly right!